android/expoview/src/main/java/host/exp/exponent/storage/ExperienceDBObject.kt

```diff
@@ -0,0 +1,14 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+package host.exp.exponent.storage
+
+import com.raizlabs.android.dbflow.annotation.Column
+import com.raizlabs.android.dbflow.annotation.PrimaryKey
+import com.raizlabs.android.dbflow.annotation.Table
+
+@Table(database = ExponentDB::class)
+class ExperienceDBObject(
+  @PrimaryKey @Column(name = "id") var scopeKey: String? = null,
+  @Column var manifestUrl: String? = null,
+  @Column var bundleUrl: String? = null,
+  @Column var manifest: String? = null
```

Comment:
All of these values that get passed in a non-null -- prefer making these properties non-nullable if possible

Reply:
From https://dbflow.gitbook.io/dbflow/usage2/usage/models#columns:

> In Kotlin, Column properties must be public and var for now. In future versions, we hope to support Kotlin constructors without default arguments. For now, all must be var and provide a default constructor. We respect nullability of the properties and won't assign null to them if they're not nullable, but they must provide a default value.

Rather than make these default to empty string which seems more incorrect I prefer to just let them be nullable and then add non-null assertions at the callsites. When they change to allowing val, we can upgrade and remove the nullability/mutability.

Reply:
Hm, what about using lateinit?

Reply:
Nope, unfortunately lateinit isn't allowed on constructor parameters since it's explicitly designed for things that can't be initialized in the constructor.